sensibo/StateHandler.js

```diff
@@ -131,22 +129,12 @@ module.exports = (device, platform) => {
 				}
 
 				const sensiboNewACState = unified.sensiboFormattedACState(device, state)
-				const sensiboNewClimateReactState = unified.sensiboFormattedClimateReactState(device, state)
 
 				log.easyDebug(device.name, ' -> Setting New State:')
 				log.easyDebug(JSON.stringify(sensiboNewACState, null, 0))
 
-				if (platform.enableClimateReactAutoSetup) {
-					log.easyDebug(JSON.stringify(sensiboNewClimateReactState, null, 0))
-				}
-
 				try {
 					// send state command to Sensibo
-
-					if (platform.enableClimateReactAutoSetup) {
-						await sensiboApi.setDeviceClimateReactState(device.id, sensiboNewClimateReactState)
```

Comment:
If I understood your question correctly, the code that sets device.state.smartMode.enabled to true, is found in StateManager.js:

```
			// CLIMATE REACT
			ClimateReactEnabledSwitch: (state, callback) => {
				log.easyDebug(device.name, '(SET) - Climate React Enabled Switch:', state)
				const smartModeState = device.state.smartMode

				smartModeState.enabled = state

				// NOTE: we must set the 'smartMode' property directly (and NOT for example like so: device.state.smartMode.enabled = true),
				//       otherwise the StateHandler's setter code will not be executed and any changes will not take effect.
				device.state.smartMode = smartModeState

				callback()
			},
```

When the user toggles the switch device.state.smartMode is updated such that device.state.smartMode is true (or false, depending on the toggle). That, in return, triggers the following code in StateHandler.js:

```
			// Send Climate React state command and refresh state
			if (prop === 'smartMode') {
				try {
					log.easyDebug(`${device.name} - Setting Climate React state to ${value}`)
					const sensiboNewClimateReactState = unified.sensiboFormattedClimateReactState(device, state)

					sensiboApi.setDeviceClimateReactState(device.id, sensiboNewClimateReactState)
				} catch(err) {
					log('Error occurred! -> Climate React state did not change')
				}

				if (!platform.setProcessing) {
					platform.refreshState()
				} else {
					log.easyDebug('setProcessing is set to true, skipping state refresh due to Climate React set.')
				}

				return
			}
```

Which calls sensiboApi.setDeviceClimateReactState that posts new Climate React settings to Sensibo's API. The same happens when any relevant Climate React setting is set via calling the updateClimateReact function. 

For example, when setting CoolingThresholdTemperature (in StateManager.js) we then call updateClimateReact(device) which in turns triggers the aforementioned code in StateHandler.js the ends up triggering sensiboApi.setDeviceClimateReactState and posts the new Climate React settings to Sensibo's API.

Makes sense?

Reply:
No worries. Yes, you understood correctly.

There's no need to remove the return (it was there before I got here btw), since when activating Climate React does not necessitate updating the AC state. If necessary, Climate React will update the AC state on its own.

Hope that clears things up.